drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -0,0 +1,603 @@
+/*
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <string.h>
+#include <sys/byteorder.h>
+
+#include "bq274xx.h"
+
+#if defined(DT_INST_0_TI_BQ274XX_DESIGN_VOLTAGE)
+#define BQ274XX_DESIGN_VOLTAGE		DT_INST_0_TI_BQ274XX_DESIGN_VOLTAGE
+#else
+#define BQ274XX_DESIGN_VOLTAGE		3700
+#endif
+#if defined(DT_INST_0_TI_BQ274XX_DESIGN_CAPACITY)
+#define BQ274XX_DESIGN_CAPACITY		DT_INST_0_TI_BQ274XX_DESIGN_CAPACITY
+#else
+#define BQ274XX_DESIGN_CAPACITY		1800
+#endif
+#if defined(DT_INST_0_TI_BQ274XX_TAPER_CURRENT)
+#define BQ274XX_TAPER_CURRENT		DT_INST_0_TI_BQ274XX_TAPER_CURRENT
+#else
+#define BQ274XX_TAPER_CURRENT		45
+#endif
+#if defined(DT_INST_0_TI_BQ274XX_TERMINATE_VOLTAGE)
+#define BQ274XX_TERMINATE_VOLTAGE	DT_INST_0_TI_BQ274XX_TERMINATE_VOLTAGE
+#else
+#define BQ274XX_TERMINATE_VOLTAGE	3000
+#endif
+
+static int bq274xx_command_reg_read(struct bq274xx_data *bq274xx, u8_t reg_addr,
+					s16_t *val)
+{
+	u8_t i2c_data[2];
+	int status;
+
+	status = i2c_burst_read(bq274xx->i2c, BQ274XX_I2C_ADDRESS,
+			reg_addr, i2c_data, 2);
+	if (status < 0) {
+		printk("Unable to read register\n");
+		return -EIO;
+	}
+
+	*val = (i2c_data[1] << 8) | i2c_data[0];
+
+	return 0;
+}
+
+static int bq274xx_control_reg_write(struct bq274xx_data *bq274xx, u16_t subcommand)
+{
+	u8_t i2c_data, reg_addr;
+	int status = 0;
+
+	reg_addr = BQ274XX_COMMAND_CONTROL_LOW;
+	i2c_data = (u8_t)((subcommand) & 0x00FF);
+
+	status = i2c_reg_write_byte(bq274xx->i2c, BQ274XX_I2C_ADDRESS, reg_addr, i2c_data);
+	if (status < 0) {
+		printk("Failed to write into control low register\n");
+		return -EIO;
+	}
+
+	k_sleep(5);
```

Comment:
Constants should be defined and documented if applicable (e.g. is `5` a recommended value by vendor?)

Reply:
Yes, delay in needed to wait for the data to be stable. Moved to macro. Thanks.